rosbag2_cpp/include/rosbag2_cpp/writers/sequential_writer.hpp

```diff
@@ -164,7 +164,7 @@ class ROSBAG2_CPP_PUBLIC SequentialWriter
   std::unordered_map<std::string, rosbag2_storage::TopicInformation> topics_names_to_info_;
   std::mutex topics_info_mutex_;
 
-  LocalMessageDefinitionSource message_definitions_;
+  std::unique_ptr<LocalMessageDefinitionSource> message_definitions_;
```

Comment:
There are no benefits to making `message_definitions_` a unique pointer and resetting it on each writer's open/close operations. I would say doing so will slow down consequent write operations to the newly open writer in comparison to the original implementation.
Please return it back to reduce unrelated delta.

Reply:
The message_definitions_  was addressed in the commit https://github.com/ros2/rosbag2/pull/1599/commits/5a1df9e50c6b10964b4b73951613b5f7b4a8c144